File: 392. Is Subsequence/392. Is Subsequence.java
class Solution {

  public boolean isSubsequence(String s, String t) {
    if (s.isEmpty()) {
      return true;
    }

    int n = s.length();
    int m = t.length();

    if (n > m) {
      return false;
    }

    int i = 0;
    for (int j = 0; j < m; j++) {
      if (s.charAt(i) == t.charAt(j)) {
        i++;
      }

      if (i == n) {
        return true;
      }
    }

    return false;
  }
}

File: 392. Is Subsequence/392. Is Subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if s == "":
            return True

        n = len(s)
        m = len(t)

        if n > m:
            return False

        i = 0
        for j in range(m):
            if s[i] == t[j]:
                i += 1

            if i == n:
                return True

        return False
